lib/runners/processor/clang_tidy.rb

```diff
@@ -0,0 +1,99 @@
+module Runners
+  class Processor::ClangTidy < Processor
+    Schema = StrongJSON.new do
+      let :runner_config, Schema::BaseConfig.base.update_fields { |fields|
+        fields.merge!({
+          apt: enum?(string, array(string)),
+          'include-path': enum?(string, array(string))
+        })
+      }
+
+      let :issue, object(
+        severity: string,
+      )
+    end
+
+    register_config_schema(name: :clang_tidy, schema: Schema.runner_config)
+
+    # @see https://github.com/github/linguist/blob/775b07d40c04ef6e0051a541886a8f1e30a892f4/lib/linguist/languages.yml#L532-L535
+    # @see https://github.com/github/linguist/blob/775b07d40c04ef6e0051a541886a8f1e30a892f4/lib/linguist/languages.yml#L568-L584
+    VALID_EXTENSIONS = [".c", ".cpp", ".c++", ".cc", ".cp", ".cxx"].freeze
+    GLOB_HEADERS = "**/*.{h,h++,hh,hpp,hxx,inc,inl,ipp,tcc,tpp}".freeze
+
+    def setup
+      deploy_packages
+      yield
+    end
+
+    def analyzer_bin
+      "clang-tidy"
+    end
+
+    def analyze(changes)
+      issues = []
+
+      changes
+        .changed_paths
+        .select { |path| VALID_EXTENSIONS.include?(path.extname.downcase) }
```

Comment:
[suggestion] We can use also glob like GLOB_HEADERS via `fnmatch?`:

```ruby
GLOB_SOURCES = "*.{c,cpp,c++,cc,cp,cxx}".freeze
```

```suggestion
        .select { |path| path.fnmatch?(GLOB_SOURCES, File::FNM_EXTGLOB | File::FNM_CASEFOLD) }
```

See also:

- https://ruby-doc.org/stdlib/libdoc/pathname/rdoc/Pathname.html#fnmatch-3F-method
- https://ruby-doc.org/core/File.html#fnmatch-method

Reply:
LGTM.
commit c142082debb6e8e4ff1a0143c94519647018c232